controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/DeleteScopeTask.java

```diff
@@ -0,0 +1,138 @@
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static io.pravega.shared.NameUtils.READER_GROUP_STREAM_PREFIX;
+
+/**
+ * Request handler for executing a delete operation for a ReaderGroup.
+ */
+public class DeleteScopeTask implements ScopeTask<DeleteScopeEvent> {
+    private static final TagLogger log = new TagLogger(LoggerFactory.getLogger(DeleteScopeTask.class));
+
+    private final StreamMetadataStore streamMetadataStore;
+    private final StreamMetadataTasks streamMetadataTasks;
+    private final KVTableMetadataStore kvtMetadataStore;
+    private final ScheduledExecutorService executor;
+
+    public DeleteScopeTask(final StreamMetadataTasks streamMetadataTasks,
+                           final StreamMetadataStore streamMetaStore,
+                           final KVTableMetadataStore kvtMetadataStore,
+                           final ScheduledExecutorService executor) {
+        Preconditions.checkNotNull(streamMetaStore);
+        Preconditions.checkNotNull(executor);
+        this.streamMetadataStore = streamMetaStore;
+        this.streamMetadataTasks = streamMetadataTasks;
+        this.kvtMetadataStore = kvtMetadataStore;
+        this.executor = executor;
+    }
+
+    @Override
+    public CompletableFuture<Void> execute(final DeleteScopeEvent request) {
+        String scope = request.getScope();
+        long requestId = request.getRequestId();
+        final OperationContext context = streamMetadataStore.createScopeContext(scope, requestId);
+        log.debug(requestId, "Deleting {} scope", scope);
+        return streamMetadataStore.checkScopeExists(scope, context, executor).thenCompose( exists -> {
+            if (exists) {
+                RetryHelper.withRetriesAsync(() -> deleteScopeContent(scope, context, requestId)
+                        .thenCompose(table -> streamMetadataStore.deleteScopeRecursive(scope, context, executor)),
+                        e -> Exceptions.unwrap(e) instanceof RetryableException, Integer.MAX_VALUE, executor);
+                return CompletableFuture.completedFuture(null);
+            }
+            return CompletableFuture.completedFuture(null);
+        });
+    }
+
+    public CompletableFuture<Void> deleteScopeContent(String scopeName, OperationContext context, long requestId) {
+        List<String> readerGroupList = new ArrayList<>();
+        Iterator<Stream> iterator = listStreams(scopeName, context).asIterator();
+
+        // Seal and delete streams and add entry to RGList
+        while (iterator.hasNext()) {
+            Stream stream = iterator.next();
+            if (stream.getStreamName().startsWith(READER_GROUP_STREAM_PREFIX)) {
+                readerGroupList.add(stream.getStreamName().substring(
+                        READER_GROUP_STREAM_PREFIX.length()));
+            }
+            streamMetadataTasks.sealStream(scopeName, stream.getStreamName(), requestId)
```

Comment:
The future returned here is never used.

Reply:
@shshashwat you need to use `Futures.getThrowingException` as done inside StreamManagerImpl.deleteScope(scope,force=true)

Reply:
`Futures.getThrowingException` is added now before the seal and delete stream